hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -93,10 +96,15 @@ public void validateSemantics(
             @NonNull final WritableAccountStore accountStore,
             @NonNull final ReadableTokenStore tokenStore,
             @NonNull final ReadableTokenRelationStore tokenRelStore,
-            @NonNull final ReadableNftStore nftStore) {
+            @NonNull final ReadableNftStore nftStore,
+            @NonNull final ReadableAirdropStore airdropStore) {
         var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
         validateTrue(
                 op.tokenTransfers().size() <= tokensConfig.maxAllowedAirdropTransfersPerTx(), INVALID_TRANSACTION_BODY);
+        validateTrue(
+                airdropStore.sizeOfState() + roughAirdropCount(op.tokenTransfers())
+                        <= tokensConfig.maxAllowedPendingAirdrops(),
+                MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
```

Comment:
We should probably do this after we know pendingAirdrops that will be created.
Probably after `separateFungibleTransfers` and `separateNftTransfers`

Reply:
That was an option Ivan suggested 😄 Okay, I will move it there 👍

Reply:
Cool! Thanks

Reply:
Fixed 👍